integration-tests/src/tests/client/flat_storage.rs

```diff
@@ -455,42 +456,44 @@ fn test_flat_storage_iter() {
         let shard_uid = ShardUId::from_shard_id_and_layout(shard_id, &shard_layout);
         let items: Vec<_> = store.iter(shard_uid).collect();
 
-        let shard_id: u64 = shard_id.into();
-        match shard_id {
-            0 => {
-                let expected = 2 + protocol_version_modifier;
-                assert_eq!(expected, items.len());
-                // Two entries - one for 'near' system account, the other for the contract.
-                // (with newer protocol: +1 for BandwidthSchedulerState)
-                assert_eq!(
-                    TrieKey::Account { account_id: "near".parse().unwrap() }.to_vec(),
-                    items[0].as_ref().unwrap().0.to_vec()
-                );
-            }
-            1 => {
-                // Two entries - one for account, the other for contract.
-                // (with newer protocol: +1 for BandwidthSchedulerState)
-                let expected = 2 + protocol_version_modifier;
-                assert_eq!(expected, items.len());
-                assert_eq!(
-                    TrieKey::Account { account_id: "test0".parse().unwrap() }.to_vec(),
-                    items[0].as_ref().unwrap().0.to_vec()
-                );
-            }
-            2 => {
-                // Test1 account was not created yet - so no entries.
-                // (with newer protocol: +1 for BandwidthSchedulerState)
-                let expected = 0 + protocol_version_modifier;
-                assert_eq!(expected, items.len());
-            }
-            _ => {
-                panic!("Unexpected shard_id");
-            }
+        if ![s0, s1, s2].contains(&shard_id) {
+            panic!("Unexpected shard ID: {shard_id}");
+        }
+
+        if shard_id == s0 {
+            let expected = 2 + protocol_version_modifier;
+            assert_eq!(expected, items.len());
+            // Two entries - one for 'near' system account, the other for the contract.
+            // (with newer protocol: +1 for BandwidthSchedulerState)
+            assert_eq!(
+                TrieKey::Account { account_id: "near".parse().unwrap() }.to_vec(),
+                items[0].as_ref().unwrap().0.to_vec()
+            );
+        }
+        if shard_id == s1 {
+            // Two entries - one for account, the other for contract.
+            // (with newer protocol: +1 for BandwidthSchedulerState)
+            let expected = 2 + protocol_version_modifier;
+            assert_eq!(expected, items.len());
+            assert_eq!(
+                TrieKey::Account { account_id: "test0".parse().unwrap() }.to_vec(),
+                items[0].as_ref().unwrap().0.to_vec()
+            );
+        }
+        if shard_id == s2 {
+            // Test1 account was not created yet - so no entries.
+            // (with newer protocol: +1 for BandwidthSchedulerState)
+            let expected = 0 + protocol_version_modifier;
+            assert_eq!(expected, items.len());
```

Comment:
Not related to this change: we could add an `assert_matches` on `TrieKey::BandwidthSchedulerState`